### PR TITLE
doc: linguistic changes in C++ API docs

### PR DESCRIPTION
**Pull request by mgrudzinska**
- the file description is added
- @defgroup requires 2 args - label and a name
- I changed '@return Result::Success when succeed' to '@return Result::Success' everywhere, where this is the only result that can be returned. I find it confusing as it was, since the question arises "and what if it does not succeed? "
- using @c before Result::Success doesn't look good - it is already marked as link, so do we have to change its font?
before:
docs_before
after:
docs_after

### CODE REVIEW

File: inc/thorvg.h [hunk 1]
```diff
@@ -1119,24 +1128,24 @@ class TVG_EXPORT SwCanvas final : public Canvas
      *
      * The buffer of a desirable size should be allocated and owned by the caller.
      *
-     * @param[in] buffer A pointer to a memory block of the size @p w x @p h, where the raster data are stored.
+     * @param[in] buffer A pointer to a memory block of the size @p stride x @p h, where the raster data are stored.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1131]: in general the buffer size is stride x h, right?

Comment (hermet) [reply inc/thorvg.h:1131]: @mgrudzinska correct.

File: inc/thorvg.h [hunk 2]
```diff
@@ -233,19 +240,19 @@ class TVG_EXPORT Paint
      *
      * @param[in] o The opacity value in the range [0 ~ 255], where 0 is completely transparent and 255 is opaque.
      *
-     * @return @c Result::Success.
+     * @return Result::Success.
      *
-     * @note A half-translucent paint by opacity may require multiple render pass for composition, TVG recommends to avoid opacity change as possible.
+     * @note Setting the opacity with this API may require multiple renderings using a composition. It is recommended to avoid changing the opacity if possible.
```
Comment (hermet) [inline inc/thorvg.h:245]: "render pass" is more clear.

File: inc/thorvg.h [hunk 3]
```diff
@@ -170,49 +179,47 @@ struct Matrix
 /**
  * @class Paint
  *
- * @brief A abstract class for managing graphical elements.
- *
- * Basically a graphical elements in TVG is composed as an object in a scene (Canvas).
- * Paint represents a graphical object, suggest common behaviors such as duplication, transformation and composition, etc.
- * TVG recommends user to regard a paint as a pack of volatile commands, they can prepare a paint then request Canvas to run it.
```
Comment (hermet) [inline inc/thorvg.h:177]: Do you need to remove the last sentence? It's a tip for tvg concept.

File: inc/thorvg.h [hunk 4]
```diff
@@ -258,16 +265,16 @@ class TVG_EXPORT Paint
      * @param[out] w The width of the object.
      * @param[out] h The height of the object.
      *
-     * @return @c Result::Success when succeed, @c Result::InsufficientCondition otherwise.
+     * @return Result::Success when succeed, Result::InsufficientCondition otherwise.
      *
-     * @note bounding box doesn't indicate the rendering region in the result but primitive region of the object.
+     * @note The bounding box doesn't indicate the final rendered region. It's the rectangular bounding area of the object.
```
Comment (hermet) [inline inc/thorvg.h:270]: is there any clear words to express it instead of "rectangular bounding area" ?

File: inc/thorvg.h [hunk 5]
```diff
@@ -317,9 +324,9 @@ class TVG_EXPORT Fill
      * @brief Sets the parameters of the colors of the gradient and their position.
      *
      * @param[in] colorStops An array of ColorStop data structure.
-     * @param[in] cnt The count of the @p colorStops array equal to the colors number used in the gradient.
+     * @param[in] cnt The size of the @p colorStops array equal to the colors number used in the gradient.
```
Comment (hermet) [inline inc/thorvg.h:327]: count is more clear. its parameter says "count"
size is ambiguous since  data size * data count

File: inc/thorvg.h [hunk 6]
```diff
@@ -317,9 +324,9 @@ class TVG_EXPORT Fill
...
      *
-     * @return @c Result::Success when succeed.
```
Comment (hermet) [inline inc/thorvg.h:322]: We should not fix it, it always return Result::Success.

File: inc/thorvg.h [hunk 7]
```diff
@@ -394,63 +401,64 @@ class TVG_EXPORT Canvas
      * @brief Passes drawing elements to the Canvas using Paint objects.
      *
      * Only pushed paints in the canvas will be drawing targets.
-     * They are retained by the canvas until you call Canvas::clear().
-     * If you know the count of push() in the advance, please call reserve().
+     * They are retained by the canvas until you call clear().
```
Comment (hermet) [inline inc/thorvg.h:404]: Canvas::clear()
Canvas::reserve()

File: inc/thorvg.h [hunk 8]
```diff
@@ -394,63 +401,64 @@ class TVG_EXPORT Canvas
...
+     * If a @c nullptr is passed all paint objects retained by the Canvas are updated,
+     * otherwise only the paint to which the given @p paint points.
      *
      * @param[in] paint A pointer to the Paint object or @c nullptr.
      *
-     * @return @c Result::Success when succeed, @c Result::InsufficientCondition otherwise.
+     * @return Result::Success when succeed, Result::InsufficientCondition otherwise.
      *
-     * @note Updation behavior could be asynchronous by thread option.
+     * @note Update behavior can be asynchronous based on the assigned thread number.
```
Comment (hermet) [inline inc/thorvg.h:441]: number is ambiguous...  just async/sync. either thread or not.

File: inc/thorvg.h [hunk 9]
```diff
@@ -394,63 +401,64 @@ class TVG_EXPORT Canvas
...
 
     /**
-     * @brief Request the canvas to draw paint objects.
+     * @brief Request the canvas to draw the Paint objects.
      *
-     * @return @c Result::Success when succeed, @c Result::InsufficientCondition otherwise.
+     * @return Result::Success when succeed, Result::InsufficientCondition otherwise.
      *
-     * @note Drawing could be asynchronous by thread option. To guarantee the done of drawing job, call sync().
+     * @note Drawing can be asynchronous based on the assigned thread number. To guarantee the drawing is done, call sync() afterwards.
```
Comment (hermet) [inline inc/thorvg.h:450]: same. thread number

File: inc/thorvg.h [hunk 10]
```diff
@@ -477,31 +485,31 @@ class TVG_EXPORT LinearGradient final : public Fill
...
+     * the given points (@p x1, @p y1) and (@p x2, @p y2), respectively. Both lines are perpendicular to the line linking
+     * (@p x1, @p y1) and (@p x2, @p y2).
      *
      * @param[out] x1 The horizontal coordinate of the first point used to determine the gradient bounds.
      * @param[out] y1 The vertical coordinate of the first point used to determine the gradient bounds.
      * @param[out] x2 The horizontal coordinate of the second point used to determine the gradient bounds.
      * @param[out] y2 The vertical coordinate of the second point used to determine the gradient bounds.
      *
-     * @return @c Result::Success when succeed.
+     * @return Result::Success.
```
Comment (hermet) [inline inc/thorvg.h:512]: It's open case. please keep it "when succeed". it could return any error value.

File: inc/thorvg.h [hunk 11]
```diff
@@ -1039,10 +1047,10 @@ class TVG_EXPORT Picture final : public Paint
  *
  * @brief A class to composite children paints.
  *
- * As the traditional graphics rendering method, TVG also enables scene-graph mechanism,
+ * As the traditional graphics rendering method, TVG also enables scene-graph mechanism.
  * This feature supports an array function for managing the multiple paints as one group paint.
  *
- * As a group, scene can be transformed, translucent, composited with other target paints,
+ * As a group, the scene can be transformed, made opaque and composited with other target paints,
```
Comment (hermet) [inline inc/thorvg.h:1053]: We intent to say "translucent." not opaque

**Review by hermet — CHANGES_REQUESTED** [review]
good job. Please check comments.

File: inc/thorvg.h (hunk 6 above)
Comment (hermet) [reply inc/thorvg.h:322]: all other changes are same.

File: inc/thorvg.h (hunk 7 above)
Comment (mgrudzinska) [reply inc/thorvg.h]: done

File: inc/thorvg.h (hunk 8 above)
Comment (mgrudzinska) [reply inc/thorvg.h]: "The Update behavior can be asynchronous if the assigned thread number is greater than zero.
* @see Initializer::init()"

if you prefer we can stay with 'thread option' although it is not clear to me, which option do we mean:
"The Update behavior can be asynchronous depending on the thread option."

File: inc/thorvg.h (hunk 11 above)
Comment (mgrudzinska) [reply inc/thorvg.h]: done

File: inc/thorvg.h (hunk 8 above)
Comment (hermet) [reply inc/thorvg.h]: @mgrudzinska  ""The Update behavior can be asynchronous if the assigned thread number is greater than zero." this one is much clear.

File: inc/thorvg.h (hunk 5 above)
Comment (hermet) [reply inc/thorvg.h]: @mgrudzinska then it'd rather this:
The count of the @p colorStop equal ... 
I think size can't be applied this. it requires exact number of the element. not size.

File: inc/thorvg.h (hunk 4 above)
Comment (hermet) [reply inc/thorvg.h]: @mgrudzinska great.

File: inc/thorvg.h (hunk 3 above)
Comment (hermet) [reply inc/thorvg.h:177]: @mgrudzinska volatile means commands are just one single input that turns to invalid soon. So you suggestion can't satisfy with the meaning.

File: inc/thorvg.h (hunk 2 above)
Comment (hermet) [reply inc/thorvg.h]: @mgrudzinska "render pass" "composition" are a sort of terms those are used among the graphics world.

**Review by hermet — CHANGES_REQUESTED** [review]
Please check comments.

File: inc/thorvg.h (hunk 2 above)
Comment (mgrudzinska) [reply inc/thorvg.h]: restored as it was originally: 
"Setting the opacity with this API may require multiple render pass for composition."

File: inc/thorvg.h (hunk 3 above)
Comment (mgrudzinska) [reply inc/thorvg.h:177]: restored you version:
"TVG recommends user to regard a paint as a set of volatile commands. They can prepare a Paint and then request a Canvas to run them."

File: inc/thorvg.h (hunk 5 above)
Comment (mgrudzinska) [reply inc/thorvg.h]: restored your version

**Review by hermet — APPROVED** [review]
very nice. thank you.

**Pull request merged**